frontend/talentsearch/src/js/components/search/SearchContainer.tsx

```diff
@@ -234,20 +246,52 @@ export const SearchContainer: React.FC<SearchContainerProps> = ({
           />
         </div>
         <div>
-          {!updatePending ? (
-            <div>
-              {poolCandidateResults?.map(({ pool, candidateCount }) => (
-                <CandidateResults
-                  key={pool.id}
-                  candidateCount={candidateCount}
-                  pool={pool}
-                  handleSubmit={tryHandleSubmit}
-                />
-              ))}
-            </div>
-          ) : (
-            <Spinner />
-          )}
+          {!updatePending &&
+            (poolCandidateResults && poolCandidateResults.length > 0 ? (
```

Comment:
This is just my personal opinion and I'm more than willing to relent on this because it's pretty nit-picky but, I'm not a huge fan of this condition and find it kinda awkward to read. 

Could we maybe extract this out to simplify it? @petertgiles @vd1992 

## Definition

```tsx
interface ResultsDisplay {
  pending: boolean;
  results?: Array<PoolCandidate>;
}

const ResultsDisplay = ({pending, results}: ResultsDisplayProps) => {
  if(pending) {
    return <Spinner />;
  }

  return results && results.length ? (
    // CandidateResults
  ) : (
    // We can still help message
  )
}
```

## Usage

```tsx
<ResultsDisplay pending={updatePending} results={poolCandidateResults} />
```

Reply:
Ah, yeah, I like that.  I tried to improve it with a ternary but this is better imo.  :mage:

Reply:
thanks for the implementation details
done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/4903/commits/79728bc98c4d8c4af94cf72c2e5371cb3eee8b4b